core/src/main/java/tc/oc/pgm/filters/dynamic/FilterMatchModule.java

```diff
@@ -89,10 +93,22 @@ public FilterMatchModule(Match match) {
   // Filterables that need a check in the next tick (cleared every tick)
   private final Set<Filterable<?>> dirtySet = new HashSet<>();
 
-  // We have to listen to this event to guarantee that all other modules have been loaded
-  // and that there is no unresolved xml filter references.
-  @EventHandler
-  public void onMatchLoad(MatchLoadEvent event) {
+  // See MatchImpl#load
+  public void onMatchLoad() {
+
+    // First, find and create all necessary reactors
+    this.context
+        .getAll(FilterDefinition.class)
```

Comment:
I'm pretty sure that by getting all Filters, you're already finding the children, at the very least the ones with an id because there's no way those aren't put in the context

By going over each of the filters, and then also checking for children of the filters, you're likely creating reactors multiple times for some filters